keps/sig-node/1287-in-place-update-pod-resources/README.md

```diff
@@ -881,6 +887,73 @@ Other components:
 * check how the change of meaning of resource requests influence other
   Kubernetes components.
 
+### Instrumentation
+
+The kubelet will record the following metrics:
+
+#### `kubelet_container_resize_requests_total` 
+
+This metric tracks the total number of resize requests observed by the Kubelet, counted at the container level.
+A single pod update changing multiple containers will be considered separate resize requests.
+
+Labels: 
+- `resource_type` - what type of resource is being resized. Possible values: `cpu_limits`, `cpu_requests` `memory_limits`, or `memory_requests`. If more than one of these resource types is changing in the resize request, 
+we increment the counter multiple times, once for each. This means that a single container update changing multiple
+resource types will be considered multiple requests for this metric. 
+- `operation_type` -  whether the resize is an increase or a decrease. Possible values: `increase`, `decrease`, `add`, or `remove`.
+- `namespace` - the namespace of the pod.
+
+This metric is recorded as a counter.
+
+#### `kubelet_pod_resize_sli_duration_seconds`
+This metric tracks the latency between when the kubelet accepts a resize request and when it finshes actuating the request. More precisely, this metric tracks the total amount of time that the PodResizeInProgress condition is present on a pod.
```

Comment:
I'm not sure how useful this is going to be in practice, since there are so many confounding factors. Basically everything else happening in the pod sync loop, plus the relatively coarse granularity of the PLEG. Plus, once we add the memory limit decreasing with the possibility of the pod being stuck in-progress indefinitely, the higher percentile buckets will be meaningless.

Measuring the duration of [`doPodResizeAction`](https://github.com/kubernetes/kubernetes/blob/92de70895830ea1a9c2c6554bdab4cbee7ce867d/pkg/kubelet/kuberuntime/kuberuntime_manager.go#L699) might be more useful. We already have CRI operation metrics for the specific UpdateContainerResources call duration.